src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioTaskSchedulerFactory.cs

```diff
@@ -72,5 +76,38 @@ public Task<T> ScheduleTask<T>(Func<Task<T>> taskFunc, string taskName, Cancella
                 return _queue.Value.ScheduleTask(taskFunc, taskName, cancellationToken);
             }
         }
+
+        private class JoinableTaskFactoryTaskScheduler : TaskScheduler
```

Comment:
Does this not exist as a basic helper?

Reply:
➡️ No, this is not the desired practice for switching to the main thread. I'm adding it to the list of follow-up issues and it's the basis for https://github.com/Microsoft/vs-threading/issues/332.